ci/nosetests.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+set -u
+
+EXIT_CODE=0
+
+echo "========================================="
+echo "nosetests -a '!requires_sudo' output"
+echo "========================================="
+nosetests -a '!requires_sudo' tests $(echo "${NOSEOPTS/__uuid__/$(uuidgen)}") || EXIT_CODE=$(($EXIT_CODE || $?))
```

Comment:
I'm comparing against the nosetests script we have today... what is this addition for and why? `$(echo "${NOSEOPTS/__uuid__/$(uuidgen)}")`

Reply:
specifially the mention of "uuid" is throwing me off. if you could explain where this comes from that would be great, thanks!

Reply:
There is a bug with the current script you have today, where the output of `requires_sudo` gets overwritten by `!requires_sudo`
So we need to give it a unique name instead.

NOSEOPTS contains this part `--xunit-file=nosetests.__uuid__.xml` and we perform a regex to replace `__uuid__` with the output of `$(uuidgen)`.

uuidgen is a bin that comes with debian that generates a unique id every time